packages/docgen/test/fixtures/type-annotations/array-destructuring-any-other-type/example.ts

```diff
@@ -0,0 +1,3 @@
+function fn( [ first, second ]: ( T & S ) | V ): S {
```

Comment:
This might cover it (?), but it might be worth also adding an example for

```ts
function fn( [ first, second ]: Alias ): Foo
```

Reply:
Definitely will add a test case for that, thanks!